File: day1/main.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main() {
    FILE *stream;
    char *line = NULL;
    size_t len = 0;
    ssize_t nread;

    int calorie_count = 0;
    int max_calories = 0;

    stream = fopen("input.txt", "r");
    if (stream == NULL) {
        printf("NO FILE!");
        return 1;
    }

    while ((nread = getline(&line, &len, stream)) != -1) {
        calorie_count += atoi(line);
        if (strcmp(line, "\n") == 0) {
            if (calorie_count > max_calories) max_calories = calorie_count;
            calorie_count = 0;
        }
    }

    free(line);
    fclose(stream);

    printf("%d\n", max_calories);

    return 0;
}

File: day1/main.hs
greater :: Int -> Int -> Int
greater a b = if a > b then a else b

stringToInt :: String -> Int
stringToInt str = read str :: Int

handle :: [String] -> Int -> Int -> Int
handle [] calorie_count max_calories = max_calories
handle xs calorie_count max_calories =
    if head xs /= "" then
        handle (tail xs) (calorie_count + stringToInt (head xs)) max_calories
    else
        handle (tail xs) 0 (greater calorie_count max_calories)

main = do
    input <- readFile "input.txt"
    let result = handle (lines input) 0 0
    print result
